test/validate/windows_validate.go

```diff
@@ -130,3 +156,80 @@ func azureVnetIpamIps(result []byte) (map[string]string, error) {
 	}
 	return azureVnetIpamPodIps, nil
 }
+
+// validateNodeProperties() validates windows node properties
+func validateNodeProperties(nodes *corev1.NodeList, labels map[string]string, ipCount int) error {
+	log.Print("Validating Windows Node properties")
+
+	for index := range nodes.Items {
+		nodeName := nodes.Items[index].ObjectMeta.Name
+		// check nodes status;
+		// nodes status should be ready after cluster is created
+		nodeConditions := nodes.Items[index].Status.Conditions
+		if nodeConditions[len(nodeConditions)-1].Type != corev1.NodeReady {
+			return errors.Errorf("node %s status is not ready", nodeName)
+		}
+
+		// get node labels
+		nodeLabels := nodes.Items[index].ObjectMeta.GetLabels()
+		for key := range nodeLabels {
+			if label, ok := labels[key]; ok {
+				log.Printf("label %s is correctly shown on the node %+v", key, nodeName)
+				if label != overlayClusterLabelName {
```

Comment:
This is accessing the labels in `linux_validate`.
I think `validateNodeProperties` is OS agnostic and should stay in `validate.go` and based on the cni type it should validate the labels.